lib/cea/cea608_data_channel.js

```diff
@@ -0,0 +1,720 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.cea.Cea608DataChannel');
+
+/**
+ * 608 closed captions channel.
+ */
+shaka.cea.Cea608DataChannel = class {
+  /**
+   * @param {!shaka.cea.CeaDecoder} decoder CEA-608 decoder.
+   * @param {!number} fieldNum Field number.
+   * @param {!number} chanNum Channel number.
+   */
+  constructor(decoder, fieldNum, chanNum) {
+    /**
+     * Current Caption Type.
+     * @public {!shaka.cea.Cea608DataChannel.CaptionType}
+     */
+    this.type_ = shaka.cea.Cea608DataChannel.CaptionType.NONE;
+
+    /**
+     * Text buffer. Although we have this, we don't currently emit text mode.
+     * @private @const {!shaka.cea.Cea608Memory}
+     */
+    this.text_ =
+        new shaka.cea.Cea608Memory(decoder, fieldNum, chanNum, true);
+
+    /**
+     * Displayed memory.
+     * @private {!shaka.cea.Cea608Memory}
+     */
+    this.dpm_ =
```

Comment:
I feel that dpm_ and ndm_ need better names.  By the time I got to the code below where they are used, I had forgotten what they meant.